### PR TITLE
Add set_area proc

### PR DESCRIPTION
**Pull request by unid15**

### CODE REVIEW

**Comment by ComicIronic** [comment]
Looks like it's failing from the 1 warning.

**Comment by Intigracy** [comment]
code/game/objects/items/blueprints.dm:142: warning: oldarea: variable defined but not used

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -687,3 +687,22 @@
...
+
+/turf/proc/set_area(area/A)
+	if(ispath(A))
+		var/path = A
+		A = locate(path)
+
+		if(!A)
+			A = new path
+	else if(!isarea(A))
+		return FALSE
```
Comment (PJB3005) [inline code/game/turfs/turf.dm:699]: IMO this should throw an exception instead.

**Comment by ghost** [comment]
Does this fix the bug that lighting corners don't initialise on corners of new areas with blueprints?

**Comment by PJB3005** [comment]
> Does this fix the bug that lighting corners don't initialise on corners of new areas with blueprints?

Good thing this was reported earlier!

Question: is there actual lighting bugs? Corners don't initialize until needed.

**Comment by unid15** [comment]
I think that corners aren't even included in the new area, so that's a different issue

**Comment by PJB3005** [comment]
> He didn't make it throw the exception.

God damnit.

**Comment by ComicIronic** [comment]
Exceptions are a stain on programming.

**Comment by PJB3005** [comment]
It's still a million times better than silently consuming the error and doing nothing.

**Comment by ghost** [comment]
@PJB3005 I reported it.

**Comment by ghost** [comment]
https://github.com/d3athrow/vgstation13/issues/7883

**Comment by PJB3005** [comment]
That has nothing to do with lighting.

Hooray for not being responsible for a lighting bug!

**Pull request merged**